qutip/core/environment.py

```diff
@@ -850,12 +850,14 @@ class DrudeLorentzEnvironment(BosonicEnvironment):
     """
 
     def __init__(
-        self, T: float, lam: float, gamma: float, *, tag: Any = None
+        self, T: float, lam: float, gamma: float, *, tag: Any = None,
+        Nk: int = None
```

Comment:
Thank you, it is a good solution to be able to set `Nk` here. But a few things for polishing:

- Could we please document the parameter `Nk` in the docstring above?
- In `correlation_function`, if `self.Nk` is not None, the parameter `Nk` will now be ignored. Two options: (1) Remove the `Nk` parameter there. (2) If we have that parameter in both places, if the user passes `Nk` to the function, that should always take precedence over `self.Nk`.
- And could we modify the `correlation_function` docstring accordingly?
- All of this should be changed analogously in the underdamped and Lorentzian environments.

Reply:
Thanks for the review! 

I think the underdamped doesn't need it because there we obtain the correlation function from the power spectrum. But I did include it in docstrings and modified it for the Lorentzian one.

Reply:
Oh, you are right about the underdamped of course.

About the number of parameters, I was thinking about that as well and I am a bit unsure. But I guess if we know exactly which parameters we accept, it is better not to use `kwargs`, even if the parameter list becomes long. Because with `kwargs`, we don't have autocomplete / information from the IDE. (Also, I don't know how to make the documentation look nice)

Reply:
In that case I'll just undo the kwargs bit. I just thought it would be nice to have because codeclimate complains about it constantly. 

The only other bit that I wanted to modify while updating the tutorials is the fact that the from_spectral_density constructor (correlation function and power spectrum as well), accept a function with only one parameter w (or t) and I thought it would be nice if they have an args argument, so that we could pass the other parameters to it.

Something like 

def ohmic_spectral_density(w, alpha, wc):
    """The Ohmic bath spectral density as a function of w
    (and the bath parameters).
    """
    return w * alpha * np.exp(-w / wc)

sd_env2=BosonicEnvironment.from_spectral_density(J=ohmic_spectral_density,wlist=w,args=(alpha,wc))

Though again not sure, if that's wanted. I find that to be nice though

I'm pretty much done with the tutorials, I guess that If I go through the User's guide and don't have anythinf to add I should just mark this as "ready for review" so that it is no longer a draft PR?

Reply:
Adding an args parameter there would be nice! :)

> I'm pretty much done with the tutorials, I guess that If I go through the User's guide and don't have anythinf to add I should just mark this as "ready for review" so that it is no longer a draft PR?

Sounds good 🥳